src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClient.java

```diff
@@ -421,6 +421,36 @@ public void deleteTemplate(String projectId, String templateName) throws IOExcep
         return instanceTemplates;
     }
 
+    /**
+     * Creates persistent disk snapshot for Compute Engine instance.
+     * This method blocks until the operation completes.
+     *
+     * @param projectId
+     * @param zone
+     * @param instanceId
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public Operation.Error createSnapshot(String projectId, String zone, String instanceId) throws IOException, InterruptedException{
+        //TODO: cleanup in the case of a failed snapshot creation (try catch?)
+        // TODO: how to handle these exceptions?
+        // can catch the exceptions, but i want a case where any exception of op.getError() is non null
+        //cause CreateSnapshot extends ComputeRequest<Operation>
+        zone = nameFromSelfLink(zone);
+        //TODO remove later since these were for debugging
+        System.out.println("zone is " + zone);
```

Comment:
Remove debug printlns.

Reply:
will be gone by next PR/CR (when I finish all my development)

Reply:
What you mean by next PR/CR? if next commit then ok.. but I suppose it should be done in this PR in full..